packages/i18n/src/index.js

```diff
@@ -90,7 +95,16 @@ function dcnpgettext( domain = 'default', context, single, plural, number ) {
  * @return {string} Translated text.
  */
 export function __( text, domain ) {
-	return dcnpgettext( domain, undefined, text );
+	const translation = dcnpgettext( domain, undefined, text );
+
+	/**
+	 * Filters text with its translation.
+	 *
+	 * @param {string} translation Translated text.
+	 * @param {string} text        Text to translate.
+	 * @param {string} domain      Text domain. Unique identifier for retrieving translated strings.
+	 */
+	return applyFilters( 'i18n.gettext', translation, text, domain );
```

Comment:
@aduth - we probably should also optimize `applyFilters` to return early when there are no hooks to process. See:

https://github.com/WordPress/gutenberg/blob/master/packages/hooks/src/createRunHook.js#L22-L33

It makes me think that having to run filters on every re-render might be bad for the performance of the application.

Reply:
Okey, I see https://github.com/WordPress/gutenberg/pull/12517#issuecomment-444561526 which aligns with my comment.

Reply:
Yeah, it's not too horrible though. Outside of some conditions and a one-time setup, it amounts to incrementing the `runs` each time it's called.